platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -353,10 +353,12 @@ public void removeApplication(String appName) {
             bundleLocations.add(info.getLocation());
           }
 
-          for (Bundle bundle : context.getBundles()) {
-            for (BundleInfo info : bundles) {
-              if (info.getLocation().equals(bundle.getLocation())) {
-                metatypeInformation.add(metatypeService.getMetaTypeInformation(bundle));
+          if (context != null) {
```

Comment:
✏️  The `getContext()` method should be tagged with `@Nullable` since it can return null.

Reply:
@paouelle As per @clockard's suggestion, I will have `getContext()` throw an IllegalStateException and take out these `if`s